### ISSUE TITLE
Changes to Item Template Fields Are Not Copied To Variants

### ISSUE THREAD

**Issue by legolas108** [frappe/webshop#62]
When creating a new item and a variant based on attribute value, a change to the value of any template item field with the `No Copy` check box cleared won't be copied to the variant as it should. Under “Item Variant Settings” the “Do not update variants on save” check box is cleared and the changed field is listed in the “Copy Fields to Variant” Fields table. Uninstalling the `webshop` app "solves" the issue.

**Comment by simran-baliyan** [comment]
+1 Yes , I am also facing the same issue in V15

**Comment by GeoCommander** [comment]
I also noticed this issue

**Comment by dinoweb** [comment]
Hi, any news about this? We have the same problem here.
When webshop app is installed, no field is copied on updating the template.
Uninstalling the webshop app "solves" the issue.

**Comment by GeoCommander** [comment]
> Hi, any news about this? We have the same problem here.
> 
> When webshop app is installed, no field is copied on updating the template.
> 
> Uninstalling the webshop app "solves" the issue.  
> 
> 

yes this is super annoying. I ended up looking for other solutions. I am trying Vendure right now... looks promissing

**Comment by GeoCommander** [comment]
This is biggest issue Webshop and ERPNext facing right now @rohitwaghchaure

Frappe Webshop is way better solution than anything out there but because **item variants cant be updated** its huge problem. its been like this for past 4 month

**Comment by legolas108** [comment]
Finally found time to follow-up on this and may have found the fix.

In `webshop/webshop/doctype/override_doctype/item.py` the `on_update` method doesn't call the `super` method as the other overridden methods do. This skips the update of the variants and the item price.

Inserting the call to the `super` method has the variants updated as expected:

``` python
class WebshopItem(Item):
	def on_update(self):
		# next line inserted to fix the bug
		super(WebshopItem, self).on_update()
		invalidate_cache_for_item(self)
```

Will try to create bug report and PR now.
